src/main/scala/org/neo4j/spark/utils/Neo4jUtils.scala

```diff
@@ -0,0 +1,58 @@
+package org.neo4j.spark.utils
+import java.util.concurrent.Callable
+import java.util.function
+
+import org.neo4j.driver.{Driver, Result, Session, Transaction}
+import io.github.resilience4j.retry.{Retry, RetryConfig}
+import org.neo4j.driver.exceptions.{ServiceUnavailableException, SessionExpiredException, TransientException}
+import org.neo4j.spark.Neo4jConfig
+
+
+object Neo4jUtils {
+
+  def close(driver: Driver, session: Session): Unit = {
+    try {
+      if (session != null && session.isOpen) {
+        session.close()
+      }
+    } finally {
+      if (driver != null) {
+        driver.close()
+      }
+    }
+  }
+
+  private val retryConfig = RetryConfig.custom.retryExceptions(
+      classOf[SessionExpiredException], classOf[ServiceUnavailableException] // retry on the same exceptions the driver does [1]
+    )
+    .retryOnException(new function.Predicate[Throwable] {
+      override def test(exception: Throwable): Boolean = if (exception.isInstanceOf[TransientException]) {
```

Comment:
should probably a pattern match in the future

Reply:
Added in this one